### PR TITLE
Persist role credentials across jobs; provide db_identifier in main.tf

### PR DESCRIPTION
**Pull request by louiseswift**

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -79,16 +79,17 @@ module "postgres_db_development" {
...
   environment_name = "development"
   vpc_id = data.aws_vpc.development_vpc.id
-  db_name = "mosaic-mirror"
-  db_port  = 5002
-  subnet_ids = data.aws_subnet_ids.development_private_subnets.ids
   db_engine = "postgres"
   db_engine_version = "12.2"
+  db_identifier = "mosaic-mirror"
   db_instance_class = "db.t2.micro"
+  db_name = "mosaic-mirror"
```
Comment (louiseswift) [inline terraform/main.tf:86]: `db_identifier` variable still now required, as per: https://github.com/LBHackney-IT/aws-hackney-common-terraform/commit/b9c6300e3734cea3a102f136ecdc43c9721c9177#diff-747a367b56b74b6e938e5119f00436c5R19

Checking now whether `db_name` still needed here.

File: .circleci/config.yml [hunk 2]
```diff
@@ -138,105 +157,110 @@ jobs:
...
+              only: master
+      - assume-role-staging:
+          context: api-assume-role-development-context
+          requires:
+            - build-and-test
+          filters:
+            branches:
+              only: master
+      - terraform-init-and-apply-to-staging:
+          # context: api-assume-role-staging-context # needed if creds file persisted?
```
Comment (louiseswift) [inline .circleci/config.yml:222]: I can test this before these changes are merged, and/or watch the CI build after merge to examine the results of these jobs that I am assuming (based on behaviour in the development-only jobs) will now work without needing the credentials provided by the context.

File: .circleci/config.yml [hunk 3]
```diff
@@ -7,14 +7,21 @@ orbs:
...
   docker-python:
-      docker:
-        - image: circleci/python:3.7
+    docker:
+      - image: circleci/python:3.7
   docker-terraform:
-     docker:
-       -  image: "hashicorp/terraform:light"
+    docker:
+      - image: "hashicorp/terraform:light"
```
Comment (louiseswift) [inline .circleci/config.yml:14]: My VSCode clearly has a lot of opinions about yml formatting - let me know if problematic and I will restore the original formatting.

File: .circleci/config.yml [hunk 4]
```diff
@@ -138,105 +157,110 @@ jobs:
...
-           filters:
-             branches:
-               only: master
+    jobs:
+      - build-and-test:
+          filters:
+            branches:
+              only: master
+      - assume-role-staging:
+          context: api-assume-role-development-context
```
Comment (elcorbs) [inline .circleci/config.yml:215]: Should this not be using the staging credentials?

Comment (louiseswift) [reply .circleci/config.yml:215]: Good spot - thank you!

File: terraform/main.tf [hunk 5]
```diff
@@ -79,16 +79,17 @@ module "postgres_db_development" {
...
+  db_port  = 5002
+  db_username = "${local.parameter_store}/mosaic-postgres-username-development"
```
Comment (elcorbs) [inline terraform/main.tf:88]: Is this referencing to an ssm parameter?

File: terraform/main.tf (hunk 1 above)
Comment (louiseswift) [reply terraform/main.tf:86]: (Looks like `db_name` still [very important](https://github.com/LBHackney-IT/aws-hackney-common-terraform/blob/b9c6300e3734cea3a102f136ecdc43c9721c9177/modules/database/postgres/main.tf#L32) so leaving in place here.)

File: terraform/main.tf (hunk 5 above)
Comment (louiseswift) [reply terraform/main.tf:88]: Can confirm a param exists in SSM with this name, in the Development APIs account.

Looks like line 18 creates the reference to the correct bit of Parameter Store.

Comment (elcorbs) [reply terraform/main.tf:88]: I've been naming the SSM parameters with the convention `mosaic-api/staging/db_name`, I'm not terribly attached to this so could switch to how this one is named but should pick one or the other. What do you think?

Comment (louiseswift) [reply terraform/main.tf:88]: I like `mosaic-api/staging/db_name` (or `db-name`). I had to add `/mosaic-api/development/postgres-hostname` for testing credentials with `terraform plan` (well, seeing the error about a missing parameter was a sufficient result there, but wanted to see if `plan` could succeed also) so it looks like that convention is in half in place already.

I'll update `mosaic-postgres-username-development` to match. 👍

Comment (elcorbs) [reply terraform/main.tf:88]: nice 👍 I can clean up the other parameters which don't match it as well

**Pull request merged**